src/openloco/window.cpp

```diff
@@ -117,6 +117,40 @@ namespace openloco::ui
             viewports[1]->render(dpi);
     }
 
+    // 0x0045FCE6
+    // Input:
+    // regs.ax:  x
+    // regs.bx:  y
+    // regs.bp:  z
+    // Output:
+    // {x: regs.ax, y: regs.bx}
+    std::optional<map::map_pos> screenGetMapXyWithZ(const viewport_pos& vpos, const int16_t z)
```

Comment:
make this take an `xy32` instead of a `viewport_pos`. The caller gives it a physical mouse coordinate. This is then converted into a viewport position which is finally converted into a map position.

Reply:
I had to modify also the containsUI, you can review it in the later commit.